### DISCUSSION TITLE
Layout bars overrflowing each other

### DISCUSSION THREAD

**Discussion by NeoCoderMatrix86** [Megabit/Blazorise#3580] [Q&A]
Hello,

I try to update Blazorise from 0.9.3.5 and it works really good, but I have a little problem: I use 2 Bars in my project. One coming from MainLayout.razor and another included in Index.razor with headers fixed. So it looks (on 0.9.3.5) this way:

![grafik](https://user-images.githubusercontent.com/40752681/158129481-17bb25d5-f18f-433d-845c-62407172ea10.png)

The basic setup is like this:

MainLayout.razor:
```
<Layout>
    <LayoutHeader>
        <Bar Background="Background.Primary" ThemeContrast="ThemeContrast.Light">
[...]
       </Bar>
    </LayoutHeader>
    <LayoutContent>
        @Body
    </LayoutContent>
</Layout>
```

Index.razor:
```
<Layout>
    <LayoutHeader Fixed="true">
        <Bar Breakpoint="Breakpoint.Desktop" Background="Background.Dark" ThemeContrast="ThemeContrast.Dark">
            <BarToggler />
            <BarMenu>
                <BarStart>
</BarStart>
            </BarMenu>
            <BarItem>
                [...]
            </BarItem>
        </Bar>
    </LayoutHeader>
    <LayoutContent>
        <Row Margin="Margin.Is2">
[...]
        </Row>
    </LayoutContent>
</Layout>
```

The result is a bit broken:

![grafik](https://user-images.githubusercontent.com/40752681/158130096-de4094a5-3a6a-4e7d-9275-be98b058245c.png)

Can someone please tell me, where I was wrong? I'm currently missing a start point for research.
Thank you :)

**Comment by stsrki** [comment]
You're probably inheriting from the same Blazor layout. You can force it with `@layout MainLayout` at the beginning of the razor file.

**Reply by NeoCoderMatrix86** [reply]
If I understand you correctly, this should work, correct?

![grafik](https://user-images.githubusercontent.com/40752681/158207904-0f482861-41ce-4f8e-ac97-d4f66ffe32e2.png)

But the result is as above, the 2 bars are overflowing :(.

**Reply by stsrki** [reply]
You also have a fixed bar `<LayoutHeader Fixed="true">`. Whenever you have any fixed element on a page you're basically removing it from regular page flow. 

It is also mentioned in the BS documentation https://getbootstrap.com/docs/5.1/components/navbar/#placement

**Reply by NeoCoderMatrix86** [reply]
Ok, thanks for the documentation :). Is there any possibility to have the second LayoutHeader "sticky top" with the blazorise properties? Or should I use the bootstrap classes?

**Reply by stsrki** [reply]
You will probably need to adjust the offset with CSS or inline styles. By adding a top margin or top position.

**Reply by NeoCoderMatrix86** [reply]
So to just be clear. I should remove the `<LayoutHeader Fixed="true">` replace it with `<LayoutHeader>` and afterwards try to get the second bar sticky? Do you have any examples? Sorry, I'm not that familiar with bootstrap and design.

**Reply by stsrki** [reply]
Yes, I would remove `<LayoutHeader Fixed="true">` as that would return the fixed element to its original place on-page. Later you can try messing with CSS but be aware that you can still break it up if not careful

- https://www.w3schools.com/howto/howto_js_navbar_sticky.asp
- https://www.codeply.com/go/oY2CyNiA7A/navbar-simple-sticky-top

**Reply by NeoCoderMatrix86** [reply]
I'm currently trying to get everything into 1 Layout with only 1 Bar above that is fixed. So I moved some parts into MainLayout.razor and added `<LayoutHeader Fixed>` but the result is a bit confusing:

![grafik](https://user-images.githubusercontent.com/40752681/159011049-5df0ea87-bbd0-45a2-9ba4-f4ea97f8fa46.png)

As you can see, the ChildContent is hidden behind the header. Why is this done? What Am I missing?

**Reply by stsrki** [reply]
It's hard to tell what is wrong without running the same code. Can you try adding Visible param?

```
<Bar @bind-Visible="@visible">
@code{
  bool visible;
}
```

**Reply by stsrki** [reply]
Also try adding height through theming https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/App.razor#L23

**Reply by NeoCoderMatrix86** [reply]
Ok, adding the visible param didn't change anything. Basically the code can be found here:

https://github.com/NeoCoderMatrix86/AudioCuesheetEditor/blob/%23199-Update-to-.NET6/AudioCuesheetEditor/Shared/MainLayout.razor

I just changed the LayoutHeader in MainLayout to be fixed and commented out the whole Layout in Index.razor.

Reading the documentation here https://blazorise.com/docs/components/layout the basic idea is a layout with header fixed:

```
<Layout>
    <LayoutHeader>
        <Bar>...</Bar>
    </LayoutHeader>
    <LayoutContent>
        Index.razor
    </LayoutContent>
</Layout>
```

**Reply by NeoCoderMatrix86** [reply]
I played around a bit and adding `<LayoutHeader Fixed Position="Position.Sticky">` fixed the problem. But I think, it is a bug, since the property Fixed already says, that I want this header to be fixed. Should I fill a bug report?
